### PR TITLE
Add plugin hook for packet_msgstring

### PR DESCRIPTION
**Pull request by sctnightcore**

### CODE REVIEW

File: src/Network/Receive.pm [hunk 1]
```diff
@@ -8889,11 +8889,17 @@ sub skill_msg {
...
+	if ($msgTable[$index]) { # show message from msgstringtable.txt
+		message "$msgTable[$index]. Value: $args->{paral}\n", "info";
 	} else {
-		warning TF("Unknown msgid:%d paral:%d. Need to update the file msgstringtable.txt (from data.grf)\n", $args->{index}, $args->{paral});
+		warning TF("Unknown msgid:%d paral:%d. Need to update the file msgstringtable.txt (from data.grf)\n", $index, $args->{paral});
 	}
+
+	Plugins::callHook('packet_msgstring', {
+		index => $index,
+		val => $args->{para1}
```
Comment (alisonrag) [inline src/Network/Receive.pm:8901]: paraL, not para1

File: src/Network/Receive.pm [hunk 2]
```diff
@@ -8889,11 +8889,17 @@ sub skill_msg {
 sub msg_string {
 	my ($self, $args) = @_;
 
-	if ($msgTable[++$args->{index}]) { # show message from msgstringtable.txt
-		message "$msgTable[$args->{index}]. Value: $args->{paral}\n", "info";
+	my $index = ++$args->{index};
+	if ($msgTable[$index]) { # show message from msgstringtable.txt
+		message "$msgTable[$index]. Value: $args->{paral}\n", "info";
```
Comment (Nipodemos) [inline src/Network/Receive.pm:8894]: i've noticed that before, it had the `++` only in one place. Both in warning and message it was displayed the index without the `++`. But now every part has the plus sign. Can you confirm if the message remains as intended?

File: src/Network/Receive.pm (hunk 1 above)
Comment (ya4ept) [reply src/Network/Receive.pm:8901]: https://github.com/OpenKore/openkore/blob/master/src/Network/Receive/ServerType0.pm#L479
we use `para1`

Comment (alisonrag) [reply src/Network/Receive.pm:8901]: so need to fix it in #8896

Comment (ya4ept) [reply src/Network/Receive.pm:8901]: what is "paral"?

**Comment by ya4ept** [comment]
how can we generate this event?

File: src/Network/Receive.pm (hunk 1 above)
Comment (Nipodemos) [reply src/Network/Receive.pm]: that #8896 doesn't exist. Is it a typo @alisonrag ?

**Comment by alisonrag** [comment]
> how can we generate this event?

the only that i can record is use teleport while teleport screen is opened or try to use skill while talking NPC, both should generate the following message:
```
"any work in progress (NPC dialog, manufacturing...)quit and try again.#"
```

File: src/Network/Receive.pm (hunk 1 above)
Comment (alisonrag) [reply src/Network/Receive.pm]: line 8896* ^^
ya4ept already fixd this

File: src/Network/Receive.pm (hunk 2 above)
Comment (alisonrag) [reply src/Network/Receive.pm:8894]: at this point $index already have the correct value

**Comment by ya4ept** [comment]
we have two similar functions [msg_string](https://github.com/OpenKore/openkore/blob/master/src/Network/Receive.pm#L8889):
```perl
sub msg_string {
	my ($self, $args) = @_;

	my $index = ++$args->{index};
	if ($msgTable[$index]) { # show message from msgstringtable.txt
		message "$msgTable[$index]. Value: $args->{paral}\n", "info";
	} else {
		warning TF("Unknown msgid:%d paral:%d. Need to update the file msgstringtable.txt (from data.grf)\n", $index, $args->{paral});
	}

	Plugins::callHook('packet_msgstring', {
		index => $index,
		val => $args->{paral}
	});
}
```
and [message_string](https://github.com/OpenKore/openkore/blob/master/src/Network/Receive.pm#L10458):
```perl
# not only for mercenaries, this is an all purpose packet !
sub message_string {
	my ($self, $args) = @_;

	if ($msgTable[++$args->{msg_id}]) { # show message from msgstringtable.txt
		warning "$msgTable[$args->{msg_id}]\n";
		$self->mercenary_off() if ($args->{msg_id} >= 1267 && $args->{msg_id} <= 1270);
	} else {
		warning TF("Unknown message_string: %s. Need to update the file msgstringtable.txt (from data.grf)\n", $args->{msg_id});
	}
}
```
 We need to leave only one

**Pull request closed without merging**